src/main/thrift/parquet.thrift

```diff
@@ -595,6 +607,62 @@ union ColumnOrder {
   1: TypeDefinedOrder TYPE_ORDER;
 }
 
+struct PageLocation {
+/** Offset of the page in the file **/
+  1: required i64 offset
+
+/**
+ * Size of the page, including header. Sum of compressed_page_size and header
+ * length
+ */
+  2: required i32 compressed_page_size
+
+/**
+ * Index within the RowGroup of the first row of the page; this means pages
+ * change on record boundaries (r = 0).
+ */
+  3: required i64 first_row_index
+}
+
+struct OffsetIndex {
+/**
+ * PageLocations, ordered by increasing PageLocation.offset. It is required
+ * that page_locations[i].first_row_index > page_locations[i+1].first_row_index.
+ */
+  1: required list<PageLocation> page_locations
+}
+
+/**
+ * Description for ColumnIndex.
+ * Each <array-field>[i] refers to the page at OffsetIndex.page_locations[i]
+ */
+struct ColumnIndex {
+/**
+ * A list of bools to determine the validity of the corresponding min and max
+ * values. If this is true, the corresponding min and max values in the next
+ * two lists are non-null.
+ */
+  1: required list<bool> null_pages
```

Comment:
I think the description is no longer accurate. If true, the page contains only null values so the corresponding entries in min_values and max_values should be ignored.